src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp

```diff
@@ -625,6 +625,16 @@ bool findSuitableParallelDimension(mlir::ArrayRef<IndexExpr> lb,
     mlir::ArrayRef<IndexExpr> ub, int64_t firstInclusiveDim,
     int64_t lastExclusiveDim, int64_t &parDim, int64_t minSize = 4);
 
+// Try to find a suitable loop index for parallelism. If found, emit
+// krnl.parallel op for that loop index and return the id of the loop index.
+// Otherwise, do nothing and return -1.
+int64_t tryCreateKrnlParallel(const onnx_mlir::KrnlBuilder &createKrnl,
+    mlir::Operation *op, std::string msg, const mlir::ValueRange &loopDef,
+    mlir::ArrayRef<IndexExpr> lbs, mlir::ArrayRef<IndexExpr> ubs,
+    int64_t firstInclusiveDim = 0, int64_t lastExclusiveDim = 2,
+    mlir::ArrayRef<int64_t> exclusiveDims = {}, int64_t minSize = 4,
+    bool doNotCreateKrnlParallel = false);
```

Comment:
Nit: double negation are not great. I would suggest `bool createKrnlParallel = true'

Also, I would now add a sentence on what loop are eligible.

```
// parID >= firstInclusiveDim && parID < exlusiveDim && parID not in exclusiveDim && tripCount(parID) >= minSize when tripCount can be determined at compile time.
```
```

Reply:
Sure, made the changes. Thanks!